### PR TITLE
Initial version of plugin commands

### PR DESCRIPTION
**Pull request by Fatme**

### CODE REVIEW

File: lib/commands/plugin/add-plugin.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+///<reference path="../../.d.ts"/>
+"use strict";
+
+export class AddPluginCommand implements ICommand {
+	constructor(private $pluginsService: IPluginsService,
+		private $errors: IErrors) { }
+	
+	execute(args: string[]): IFuture<void> {
+		if(!args[0]) {
```
Comment (teobugslayer) [inline lib/commands/plugin/add-plugin.ts]: I think this if belongs to canExecute

File: lib/commands/plugin/remove-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+///<reference path="../../.d.ts"/>
+"use strict";
+
+export class RemovePluginCommand implements ICommand {
+	constructor(private $pluginsService: IPluginsService,
+		private $errors: IErrors) { }
+	
+	execute(args: string[]): IFuture<void> {
+		if(!args[0]) {
```
Comment (teobugslayer) [inline lib/commands/plugin/remove-plugin.ts]: Consider moving this check into canExecute

File: lib/definitions/plugins.d.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+interface IPluginsService {
```
Comment (teobugslayer) [inline lib/definitions/plugins.d.ts]: Consider adding JSDocs for the interface and its members

File: lib/services/plugins-service.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		private $childProcess: IChildProcess) { }
+		
+	public get modulesDestinationPath() {
+		let platformData = this.$platformsData.getPlatformData("ios");
+		return path.join(platformData.appDestinationDirectoryPath, constants.APP_FOLDER_NAME, "tns_modules");
+	} 
+		
+	public add(pluginName: string): IFuture<void> {
+		return (() => {
+			let command = `npm install ${pluginName} --save`;
```
Comment (teobugslayer) [inline lib/services/plugins-service.ts]: Maybe it is a good idea to support --production as well. I mean, our command tns plugin add should have a --production option, which we pass to npm install. Your opinion?

File: lib/services/plugins-service.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+			this.$childProcess.exec(command, { cwd: this.$projectData.projectDir }).wait();
+			this.prepare(pluginName).wait();	
+		}).future<void>()();
+	}
+	
+	public prepare(pluginName: string): IFuture<void> {
+		return (() => {
+			// Process .js files 
+			// TODO: get correctly pluginName if plugin is installed with url or folder path
```
Comment (teobugslayer) [inline lib/services/plugins-service.ts]: Until this is implemented, you should reject such plugins.

**Comment by Fatme** [comment]
I'm closing and will reopen against correct branch.

**Pull request closed without merging**